tests/integration/test_check_downloaded_output.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import subprocess
+from os import path
+import tarfile
+from pathlib import Path
+
+import utils
+
+
+def test_check_downloaded_output(test_env):
+    """
+    Check that the bundle has all the necessities.
+
+    Process:
+    * Send new request to cachito api
+    * Send request to download appropriate bundle from cachito
+
+    Checks:
+    * Check that response code is 200
+    * Check that state is "complete"
+    * Check the downloaded data are in gzip format and valid
+    * Check that dir deps/gomod/… contains cached dependencies
+    * Check that dir app/ contains application source code
+    * Check that the same full path filename is not duplicated
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response = client.create_new_request(
+        payload={
+            "repo": test_env["download_output"]["repo"],
+            "ref": test_env["download_output"]["ref"],
+            "pkg_managers": test_env["download_output"]["pkg_managers"],
+        },
+    )
+
+    assert response.status == 201
+    response = client.wait_for_complete_request(response)
+    assert response.data["state"] == "complete"
+
+    resp = client.download_bundle(response.id)
+    assert resp.status == 200
+
+    file_name = f"download_{str(response.id)}"
+    file_name_tar = f"{file_name}.tar.gz"
+    with open(file_name_tar, "wb") as file:
+        file.write(resp.data)
+
+    process = subprocess.run(["gzip", "-t", file_name_tar])
```

Comment:
Good idea!

I'm not sure if this does the same thing, but Python has an [is_tarfile](https://docs.python.org/3/library/tarfile.html#tarfile.is_tarfile) function. If it does the same thing, it might be nice to use that instead of calling out to a command that may or may not be present on the system.

Reply:
It seems that it's doing the same thing. Changed it to use is_tarfile function.